llvm/lib/CheerpUtils/PointerPasses.cpp

```diff
@@ -471,6 +471,8 @@ bool PointerToImmutablePHIRemoval::runOnFunction(Function& F)
 			if(!(isa<BranchInst>(I) && cast<BranchInst>(I).isConditional()) && !isa<SelectInst>(I))
 				continue;
 			llvm::Value* cond = I.getOperand(0);
+			if(cond->getType()->isVectorTy())
```

Comment:
I am unsure here. @alexp-sssup what does this pass do, and are vectors actually to be ignored?

Reply:
Actually, I think I might have suggested this change, but I don't see how the condition can ever be a vector type, since the spec states that Branch instructions always takes a i1 as condition, so this should in theory be safely reverted.

Reply:
Actually, correctly myself: BranchInst are always i1, while only Select may possibly be on vectors like 4 x i1, and in that case we still might potentially be interested in performing the transformation, but currently we will not actually take advantage of it. Probably it's better to add a note that this might happens only on select + modyfing the assert to handle either cond->getType()->isIntegerTy(1) or cond->getType()->isVectorTy() && cond->getType->getElmentTy()->isIntegerTy(1) [the rest of the code should be good to go]

Reply:
Removing the if and broadening the assert to also allow for vectors does not break the code.
Testing this did bring to light an error in my checkVectorCorrectness function though.